### PR TITLE
Staging - stage to /opt instead of /mnt

### PR DESCRIPTION
**Pull request by adean20csc**
Changes to stage to /opt instead of /mnt

Also change path in local.repo because file is used with and without ansible so cannot be a var

first commit omitted some rpms ... added them back

### CODE REVIEW

**Comment by adean20csc** [comment]
```
[root@staging ~]# ansible-playbook initial_stage.yml

PLAY [Download the Kragle source repo] ****************************************

TASK: [Create the staging directories on the usb hard drive] ******************
ok: [localhost] => (item=/opt/autodeploy/projects)
ok: [localhost] => (item=/opt/autodeploy/resources)

TASK: [Determine the latest tagged versions of public source Git repos] *******
ok: [localhost] => (item={'repo': 'git://github.com/csc/kragle', 'tag': 'https://api.github.com/repos/csc/kragle/tags', 'name': 'kragle'})
ok: [localhost] => (item={'repo': 'git://github.com/csc/slimer', 'tag': 'https://api.github.com/repos/csc/slimer/tags', 'name': 'slimer'})
ok: [localhost] => (item={'repo': 'git://github.com/csc/ansible-scaleio', 'tag': 'https://api.github.com/repos/csc/ansible-scaleio/tags', 'name': 'ansible-scaleio'})

TASK: [Clone the public project repos from Git] *******************************
changed: [localhost] => (item={u'content_length': u'635', u'status': 200, u'x_github_request_id': u'470E7AF9:146C6:1A6BE82:562E2F2D', u'strict_transport_security': u'max-age=31536000; includeSubdomains; preload', u'content_location': u'https://api.github.com/repos/csc/kragle/tags', u'vary': u'Accept, Accept-Encoding', u'x_served_by': u'd594a23ec74671eba905bf91ef329026', u'access_control_allow_origin': u'*', u'last_modified': u'Thu, 27 Aug 2015 12:41:21 GMT', u'content_type': u'application/json; charset=utf-8', u'x_ratelimit_limit': u'60', u'date': u'Mon, 26 Oct 2015 13:48:29 GMT', u'x_frame_options': u'deny', u'_content_encoding': u'gzip', u'x_ratelimit_reset': u'1445870427', u'x_github_media_type': u'github.v3; format=json', u'x_xss_protection': u'1; mode=block', u'transfer_encoding': u'chunked', u'changed': False, u'x_content_type_options': u'nosniff', u'access_control_allow_credentials': u'true', u'server': u'GitHub.com', u'access_control_expose_headers': u'ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', u'json': [{u'commit': {u'url': u'https://api.github.com/repos/csc/kragle/commits/32cf7e96f841c7f8a7cb5d0a9ac9e9761796263a', u'sha': u'32cf7e96f841c7f8a7cb5d0a9ac9e9761796263a'}, u'zipball_url': u'https://api.github.com/repos/csc/kragle/zipball/v0.1.0', u'tarball_url': u'https://api.github.com/repos/csc/kragle/tarball/v0.1.0', u'name': u'v0.1.0'}, {u'commit': {u'url': u'https://api.github.com/repos/csc/kragle/commits/e36116c507a169c5a7efb2b87e51e1b47c096b57', u'sha': u'e36116c507a169c5a7efb2b87e51e1b47c096b57'}, u'zipball_url': u'https://api.github.com/repos/csc/kragle/zipball/v0.0.1', u'tarball_url': u'https://api.github.com/repos/csc/kragle/tarball/v0.0.1', u'name': u'v0.0.1'}], u'etag': u'W/"825883ced92037439608f59c8e752e79"', 'item': {'repo': 'git://github.com/csc/kragle', 'tag': 'https://api.github.com/repos/csc/kragle/tags', 'name': 'kragle'}, 'invocation': {'module_name': u'uri', 'module_args': ''}, u'redirected': False, u'content_security_policy': u"default-src 'none'", u'cache_control': u'public, max-age=60, s-maxage=60', u'x_ratelimit_remaining': u'53'})
changed: [localhost] => (item={u'content_length': u'635', u'status': 200, u'x_github_request_id': u'470E7AF9:1AD3:1CCF698:562E2F2D', u'strict_transport_security': u'max-age=31536000; includeSubdomains; preload', u'content_location': u'https://api.github.com/repos/csc/slimer/tags', u'vary': u'Accept, Accept-Encoding', u'x_served_by': u'318e55760cf7cdb40e61175a4d36cd32', u'access_control_allow_origin': u'*', u'last_modified': u'Mon, 01 Jun 2015 14:37:54 GMT', u'content_type': u'application/json; charset=utf-8', u'x_ratelimit_limit': u'60', u'date': u'Mon, 26 Oct 2015 13:48:30 GMT', u'x_frame_options': u'deny', u'_content_encoding': u'gzip', u'x_ratelimit_reset': u'1445870427', u'x_github_media_type': u'github.v3; format=json', u'x_xss_protection': u'1; mode=block', u'transfer_encoding': u'chunked', u'changed': False, u'x_content_type_options': u'nosniff', u'access_control_allow_credentials': u'true', u'server': u'GitHub.com', u'access_control_expose_headers': u'ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', u'json': [{u'commit': {u'url': u'https://api.github.com/repos/csc/slimer/commits/e49ad41c8de4185b6134988f59a32f15ab602cb2', u'sha': u'e49ad41c8de4185b6134988f59a32f15ab602cb2'}, u'zipball_url': u'https://api.github.com/repos/csc/slimer/zipball/v0.1.2', u'tarball_url': u'https://api.github.com/repos/csc/slimer/tarball/v0.1.2', u'name': u'v0.1.2'}, {u'commit': {u'url': u'https://api.github.com/repos/csc/slimer/commits/800ddb77080610c267ec776857559379c54c0afe', u'sha': u'800ddb77080610c267ec776857559379c54c0afe'}, u'zipball_url': u'https://api.github.com/repos/csc/slimer/zipball/v0.1.1', u'tarball_url': u'https://api.github.com/repos/csc/slimer/tarball/v0.1.1', u'name': u'v0.1.1'}], u'etag': u'W/"bfb97f0b6dfa0adee9a71e692145fdd9"', 'item': {'repo': 'git://github.com/csc/slimer', 'tag': 'https://api.github.com/repos/csc/slimer/tags', 'name': 'slimer'}, 'invocation': {'module_name': u'uri', 'module_args': ''}, u'redirected': False, u'content_security_policy': u"default-src 'none'", u'cache_control': u'public, max-age=60, s-maxage=60', u'x_ratelimit_remaining': u'52'})
changed: [localhost] => (item={u'content_length': u'345', u'status': 200, u'x_github_request_id': u'470E7AF9:146C7:255515D:562E2F2E', u'strict_transport_security': u'max-age=31536000; includeSubdomains; preload', u'content_location': u'https://api.github.com/repos/csc/ansible-scaleio/tags', u'vary': u'Accept, Accept-Encoding', u'x_served_by': u'b0ef53392caa42315c6206737946d931', u'access_control_allow_origin': u'*', u'last_modified': u'Fri, 07 Aug 2015 00:35:06 GMT', u'content_type': u'application/json; charset=utf-8', u'x_ratelimit_limit': u'60', u'date': u'Mon, 26 Oct 2015 13:48:30 GMT', u'x_frame_options': u'deny', u'_content_encoding': u'gzip', u'x_ratelimit_reset': u'1445870427', u'x_github_media_type': u'github.v3; format=json', u'x_xss_protection': u'1; mode=block', u'transfer_encoding': u'chunked', u'changed': False, u'x_content_type_options': u'nosniff', u'access_control_allow_credentials': u'true', u'server': u'GitHub.com', u'access_control_expose_headers': u'ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval', u'json': [{u'commit': {u'url': u'https://api.github.com/repos/csc/ansible-scaleio/commits/15aca7e868546f09bf219828662b93887041fbaf', u'sha': u'15aca7e868546f09bf219828662b93887041fbaf'}, u'zipball_url': u'https://api.github.com/repos/csc/ansible-scaleio/zipball/v0.1.1', u'tarball_url': u'https://api.github.com/repos/csc/ansible-scaleio/tarball/v0.1.1', u'name': u'v0.1.1'}], u'etag': u'W/"de1dbf582f2129ceee747a36e29e9089"', 'item': {'repo': 'git://github.com/csc/ansible-scaleio', 'tag': 'https://api.github.com/repos/csc/ansible-scaleio/tags', 'name': 'ansible-scaleio'}, 'invocation': {'module_name': u'uri', 'module_args': ''}, u'redirected': False, u'content_security_policy': u"default-src 'none'", u'cache_control': u'public, max-age=60, s-maxage=60', u'x_ratelimit_remaining': u'51'})

PLAY RECAP ********************************************************************
localhost                  : ok=3    changed=1    unreachable=0    failed=0
```

File: ansible/initial_stage.yml [hunk 1]
```diff
@@ -8,9 +10,9 @@
 
   vars:
     autodeploy_user: autodeploy
```
Comment (mtnbikenc) [inline ansible/initial_stage.yml]: Looks like `autodeploy_user` can be removed from this playbook.

**Comment by jcpowermac** [comment]
Code review :+1:

**Comment by mtnbikenc** [comment]
@adean20csc Since a new stage_resources.yml playbook was created, can the stage_usb.yml playbook be removed?

**Comment by adean20csc** [comment]
was waiting on the PR to be approved and make sure we didn't want it ...

File: ansible/stage_resources.yml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        state: present
+        extra_args: "--upgrade"
+      with_items: pip_pkgs
+      tags: pkgs
+
+    - name: Create the staging directories in /opt/autodeploy
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: "{{ autodeploy_user }}"
```
Comment (mtnbikenc) [inline ansible/stage_resources.yml]: Need to remove this `owner` parameter.

File: ansible/stage_resources.yml [hunk 3]
```diff
@@ -136,16 +131,33 @@
...
+
+    - name: Copy the downloaded projects to the usb drive
+      copy:
+        src: "{{ projects_base_path }}/{{ item }}"
+        dest: "{{ usb_projects_path }}"
+      with_items: source_projects
+      when : offline
+      tags: copy
+
+    - name: Copy the downloaded projects to the usb drive
```
Comment (mtnbikenc) [inline ansible/stage_resources.yml]: Should be `resources`.

File: ansible/stage_resources.yml [hunk 4]
```diff
@@ -26,42 +23,36 @@
...
-        - "{{ usb_docker_path }}"
-        - "{{ usb_iso_path }}"
-        - "{{ usb_packages_path }}"
-        - "{{ usb_rpm_path }}"
-        - "{{ usb_scaleio_path }}"
+      with_items: resource_dirs
       tags: paths
 
-    - name: Download the private project repo from Git
+    - name: Download the private project repo from Git tp /opt/autodeploy/projects
```
Comment (mtnbikenc) [inline ansible/stage_resources.yml]: Typo `to`

**Comment by mtnbikenc** [comment]
Package dependencies are still missing.  See NGA-511.

**Comment by mtnbikenc** [comment]
Tested through playbook kragle/ansible/main.yml through the `base` role.
Reviewed :+1:

**Comment by jcpowermac** [comment]
code review :+1:

**Pull request merged**